custom_components/tplink_router/sensor.py

```diff
@@ -115,10 +145,28 @@ def __init__(
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
-        self._attr_native_value = self.entity_description.value(self.coordinator.status)
+        if self.entity_description.sensor_type == "ipv4_status":
+            data = self.coordinator.ipv4_status
+        else:
+            data = self.coordinator.status
+
+        self._attr_native_value = self.entity_description.value(data) if data else None
+
+        # Notify Home Assistant about the state update
         self.async_write_ha_state()
 
     @property
     def available(self) -> bool:
```

Comment:
It is better to separate  TPLinkRouterSensor and TPLinkRouterSensorIPV4 to to avoid  if conditions in _handle_coordinator_update and available functions